crates/ruff_linter/src/rules/flake8_simplify/rules/enumerate_for.rs

```diff
@@ -0,0 +1,178 @@
+pub struct EnumerateForLoop {
+    index_var: String,
+}
+
+impl Violation for EnumerateForLoop {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let EnumerateForLoop { index_var } = self;
+        format!("Use enumereate() for index variable `{index_var}` in for loop")
+    }
+}
+
+/// SIM113
+pub(crate) fn use_enumerate_in_for_loop(checker: &mut Checker, stmt: &Stmt) {
+    if !checker.semantic().current_scope().kind.is_function() {
+        return;
+    }
+    let Stmt::For(ast::StmtFor { body, .. }) = stmt else {
+        return;
+    };
+
+    // Check if loop body contains a continue statement
+    if loop_possibly_continues(body) {
+        return;
+    };
+    // Check if index variable is initialized to zero prior to loop
+    let Some((prev_stmt, index_var)) = get_candidate_loop_index(checker, stmt) else {
+        return;
+    };
+
+    // Check if loop body contains an index increment statement matching `index_var
+    if body.iter().any(|stmt| is_index_increment(stmt, index_var)) {
+        let diagonastic = Diagnostic::new(
+            EnumerateForLoop {
+                index_var: checker.generator().expr(index_var),
+            },
+            TextRange::new(prev_stmt.start(), stmt.end()),
+        );
+        checker.diagnostics.push(diagonastic);
+    }
+}
+
+/// Recursively check if the `for` loop body contains a `continue` statement
+fn loop_possibly_continues(body: &[Stmt]) -> bool {
+    body.iter().any(|stmt| match stmt {
+        Stmt::Continue(_) => true,
+        Stmt::If(ast::StmtIf {
+            body,
+            elif_else_clauses,
+            ..
+        }) => {
+            loop_possibly_continues(body)
+                || elif_else_clauses
+                    .iter()
+                    .any(|clause| loop_possibly_continues(&clause.body))
+        }
+        Stmt::With(ast::StmtWith { body, .. }) => loop_possibly_continues(body),
+        Stmt::Match(ast::StmtMatch { cases, .. }) => cases
+            .iter()
+            .any(|MatchCase { body, .. }| loop_possibly_continues(body)),
+        Stmt::Try(ast::StmtTry {
+            body,
+            handlers,
+            orelse,
+            finalbody,
+            ..
+        }) => {
+            loop_possibly_continues(body)
+                || loop_possibly_continues(orelse)
+                || loop_possibly_continues(finalbody)
+                || handlers.iter().any(|handler| match handler {
+                    ExceptHandler::ExceptHandler(ast::ExceptHandlerExceptHandler {
+                        body, ..
+                    }) => loop_possibly_continues(body),
+                })
+        }
+
+        _ => false,
+    })
+}
+
+/// Check previous statement of `for` loop to find a possible index variable
+/// which is initialized to zero
+/// Ex:
+/// ```python
+/// idx = 0
+/// for item in items:
+///     ...
+/// ```
+/// Return (&Stmt, &Expr) to initialization stmt and `idx` variable
+fn get_candidate_loop_index<'a>(
```

Comment:
This way does work, but i think it's better to look the binding of the name up in the semantic model and then check if it's in the same suite as the for and set to zero.